### DISCUSSION TITLE
Read current state of another, non-changed, file at each commit?

### DISCUSSION THREAD

**Discussion by ngbrown** [newren/git-filter-repo#359] [Q&A] [answered]
When either of two files changes, I need to run a process on both files and update both of them for that commit.

I have a script that is working that is using `git filter-branch --tree-filter`. I didn't write it in `git filter-rep` initially, because I thought it couldn't handle this scenario. But maybe it can. Can I use `git restore` to get the current state of a file that didn't change in the current commit using the commit `original_id`?

**Comment by newren** [comment, marked as answer]
See the contrib/filter-repo-demos/lint-history example, particularly the lint_with_real_filenames() function and cat_file_process variable.  It uses the cat_file_process to read blobs from the git history by blob id.  You could also pass it <commit_id>:<relative/file/path> instead of <blob_id> just as well.
